content/en/docs/tasks/administer-cluster/encrypt-data.md

```diff
@@ -77,11 +77,11 @@ read that resource will fail until it is deleted or a valid decryption key is pr
 {{< table caption="Providers for Kubernetes encryption at rest" >}}
 Name | Encryption | Strength | Speed | Key Length | Other Considerations
 -----|------------|----------|-------|------------|---------------------
-`identity` | None | N/A | N/A | N/A | Resources written as-is without encryption. When set as the first provider, the resource will be decrypted as new values are written.
-`aescbc` | AES-CBC with PKCS#7 padding | Strongest | Fast | 32-byte | The recommended choice for encryption at rest but may be slightly slower than `secretbox`.
-`secretbox` | XSalsa20 and Poly1305 | Strong | Faster | 32-byte | A newer standard and may not be considered acceptable in environments that require high levels of review.
+`kms` | Uses envelope encryption scheme: Data is encrypted by data encryption keys (DEKs) using AES-CBC with PKCS#7 padding, DEKs are encrypted by key encryption keys (KEKs) according to configuration in Key Management Service (KMS) | Strongest | Fast | 32-bytes |  The recommended choice, but requires a third party tool for key management. Simplifies key rotation, with a new DEK generated for each encryption, and KEK rotation controlled by the user. [Configure the KMS provider](/docs/tasks/administer-cluster/kms-provider/)
```

Comment:
this is not yet recommended choice. it is still beta-level, and has known scale issues (https://github.com/kubernetes/kubernetes/issues/90856)

cc @enj @mikedanese

Reply:
So I admit that there are a lot of [issues](https://docs.google.com/document/d/1-WHXX_Dh_MNcJb2QJxF0gOAvLjh0fAnc3QrylWdMZJA) with KMS.  However from a security perspective, I find the rest of the encryption config to not provide meaningful protection.  The only attack that the other providers would help against is if an attacker got a hold of an etcd backup (presuming the whole disk itself is encrypted, which it should always be if you care about this stuff).

The large list of problems with KMS does make it hard to recommend as a default though, especially for large clusters where it could effectively make the cluster unavailable.